common.py

```diff
@@ -71,18 +87,20 @@ class PodInfo:
 
 @dataclass
 class TestMetadata:
-    test_case_id: TestCaseType
-    test_type: TestType
     reverse: bool
-    server: PodInfo
-    client: PodInfo
+    test_case_id: TestCaseType = field(default_factory=enum_factory(TestCaseType))
+    test_type: TestType = field(default_factory=enum_factory(TestType))
+    server: PodInfo = field(default_factory=lambda: from_dict(PodInfo, {}))
+    client: PodInfo = field(default_factory=lambda: from_dict(PodInfo, {}))
 
 
 @dataclass
 class IperfOutput:
-    tft_metadata: TestMetadata
     command: str
     result: dict
+    tft_metadata: TestMetadata = field(
+        default_factory=lambda: from_dict(TestMetadata, {})
```

Comment:
why not `lambda: TestMetaData()` ?

Reply:
Good point. I'm not sure why I did that.
I removed this and refactored the code